packages/redis/src/client.ts

```diff
@@ -0,0 +1,102 @@
+import { Cluster, Redis, type RedisOptions } from "ioredis";
+import Redlock from "redlock";
+import { env } from "./env";
+
+/* Wrapper type that ties a Redis client and Redlock instance to a specific keyspace. */
+export interface RedisWithRedlock {
+  redis: Redis | Cluster;
+  redlock: Redlock;
+}
+
+/* Singleton pattern using global for dev hot reload */
+const globalForRedis = global as unknown as {
+  redisClients: Record<string, Redis | Cluster>;
+  redlockClients: Record<string, Redlock>;
+};
+
+/**
+ * Create a Redis client for a specific keyspace.
+ * NOTE: The key MUST be empty string "" if using with bullmq, as bullmq uses its own key prefixing scheme
+ * https://docs.bullmq.io/guide/connections
+ * @param keyspace A string to prefix all Redis keys with, to avoid collisions between different parts of the app.
+ * @returns A Redis client instance.
+ */
+const createRedisClient = (keyspace: string) => {
+  // Common options for both Cluster and Redis clients
+  const redisOptions: RedisOptions = {
+    maxRetriesPerRequest: null,
+    keyPrefix: !!keyspace ? `${keyspace}:` : undefined,
+    tls: env.NODE_ENV === "production" ? {} : undefined,
+  };
+
+  const RedisClient: Redis | Cluster =
+    env.NODE_ENV === "production"
+      ? // MemoryDB cluster in deployed envs
+        new Cluster([{ host: env.REDIS_HOST, port: env.REDIS_PORT }], {
+          // To prevent errors with invalid certs: https://github.com/redis/ioredis?tab=readme-ov-file#special-note-aws-elasticache-clusters-with-tls
+          dnsLookup: (address, callback) => callback(null, address),
+          redisOptions,
+        })
```

Comment:
issue: this env var setting is abit confusing across different environments (uat/prod/staging/vapt) and creates mental load on engineers. if it's possible, could we standardise the env vars and use a function to create the actual params being passed in code?